more_itertools/more.py

```diff
@@ -1200,3 +1194,26 @@ def groupby_transform(iterable, keyfunc=None, valuefunc=None):
     valuefunc = (lambda x: x) if valuefunc is None else valuefunc
     for key, group in groupby(iterable, keyfunc):
         yield key, map(valuefunc, group)
+
+
+def context(obj):
+    """Wrap *obj*, an object that supports the context manager protocol,
+    in a ``with`` statement and then yield the resultant object.
+
+    The object's ``__enter__()`` method runs before this function yields, and
+    its ``__exit__()`` method runs after control returns to this function.
+
+    This can be used to operate on objects that can close automatically when
```

Comment:
We should warn people to put the ctx mgr in the first "in" clause lest it wobble open and closed multiple times.